src/Microsoft.Health.Fhir.Core/Features/Conformance/Models/PropertyEqualityComparer.cs

```diff
@@ -0,0 +1,50 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using EnsureThat;
+
+namespace Microsoft.Health.Fhir.Core.Features.Conformance.Models
+{
+    /// <summary>
+    /// Creates an IEqualityComparer for an object based on a number of properties.
+    /// </summary>
+    /// <typeparam name="T">Type to compare</typeparam>
+    internal class PropertyEqualityComparer<T> : IEqualityComparer<T>
+    {
+        private readonly Func<T, string>[] _propertiesToCompare;
+
+        public PropertyEqualityComparer(params Func<T, string>[] propertiesToCompare)
+        {
+            EnsureArg.IsNotNull(propertiesToCompare, nameof(propertiesToCompare));
+
+            _propertiesToCompare = propertiesToCompare;
+        }
+
+        public bool Equals(T x, T y)
+        {
+            if (x == null || y == null)
+            {
+                return false;
+            }
+
+            return _propertiesToCompare.All(property => ReferenceEquals(property(x), property(y)) || property(x) == property(y));
```

Comment:
nit: save the variable so it doesn't have to evaluate again?

Reply:
Not sure what you mean? This evaluates all specified properties on ObjA against the same properties on ObjB. The instances are passed in for comparison. `_propertiesToCompare` is an array of compiled functions (getters), so should also be fast.

Reply:
`property(x)` and `property(y)` only need to be evaluated once instead of twice in the delegate.